spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/GatewayApiController.java

```diff
@@ -51,4 +66,26 @@ public ApiKey findApiKeyEntityByApiKeyValue(@RequestParam("keyValue") String api
     public ServiceApi findServiceByApiKeyValue(@RequestParam("apiKeyValue") String apiKeyValue) {
         return serviceApiRepository.findByApiKeyValue(apiKeyValue);
     }
+
+    @GetMapping( "/api/gateways/{gatewayId}/receiveApiKeyList")
+    public ResponseEntity<GatewayUpdate> findServiceByGatewayId(@PathVariable("gatewayId") String gatewayId) {
+        Proxy proxy = proxyRepository.findById(gatewayId);
+        GatewayUpdate gatewayUpdate;
+        Timestamp now = new Timestamp(System.currentTimeMillis());
+
+        if(proxy != null){
+            ServiceApi[] serviceApis = proxy.getServiceApis().stream().toArray(ServiceApi[]::new);
+            ApiKey[] apiKeys = proxy.getServiceApis().stream().flatMap(x -> x.getApiKeys()
+                .stream()).toArray(ApiKey[]::new);
+            log.info("Successfully created API key and service API list for the gateway with the ID: " + gatewayId);
+            gatewayUpdate = new GatewayUpdate(apiKeys, serviceApis, now);
+            return new ResponseEntity<>(gatewayUpdate, HttpStatus.OK);
+        }
+        else {
+            log.info("GatewayUpdate request failed. No legal object of gateway with the ID " + gatewayId +
+                    " could be found. Sending an empty Gateway update.");
+            gatewayUpdate = new GatewayUpdate(new ApiKey[0], new ServiceApi[0], now);
```

Comment:
Sending this empty object does not make much sense. Maybe simply send an empty body?